src/generators/mandoc/index.mjs

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+import { optionToMandoc, envToMandoc } from './utils/converter.mjs';
+import { writeFile, readFile } from 'node:fs/promises';
+import { join } from 'node:path';
+
+/**
+ * This generator generates a mandoc version of the API docs
+ *
+ * @typedef {Array<ApiDocMetadataEntry>} Input
+ *
+ * @type {import('../types.d.ts').GeneratorMetadata<Input, string>}
+ */
+export default {
+  name: 'mandoc',
+
+  version: '1.0.0',
+
+  description: 'Generates the `node.1` file.',
+
+  dependsOn: 'ast',
+
+  async generate(input, options) {
+    // Find the appropriate headers
+    const optionsStart = input.findIndex(({ slug }) => slug === 'options');
+    const environmentStart = input.findIndex(
+      ({ slug }) => slug === 'environment-variables-1'
+    );
+
+    // Generate the option mandoc
+    let optionsOutput = '';
+    for (let i = optionsStart + 1; i < environmentStart; i++) {
+      const el = input[i];
+      if (el.heading?.depth === 3) {
+        optionsOutput += optionToMandoc(el);
+      }
+    }
+
+    // Generate the environment mandoc
+    let envOutput = '';
+    for (let i = environmentStart + 1; i < input.length; i++) {
+      const el = input[i];
+      if (el.heading?.depth === 3) {
+        envOutput += envToMandoc(el);
+      }
+      if (el.heading?.depth < 3) break;
+    }
+
+    const apiTemplate = await readFile(
+      join(import.meta.dirname, 'template.1'),
+      'utf-8'
+    );
+    const template = apiTemplate
+      .replace('__OPTIONS__', optionsOutput)
+      .replace('__ENVIRONMENT__', envOutput);
+
+    return await writeFile(options.output, template);
```

Comment:
Is a return statement here needed?

In general you only want a return statement, if you want other generators to be able to extend on this one and use its output, which I doubt is the case.